tasks/kernels.py

```diff
@@ -0,0 +1,131 @@
+    result_dict = {}
+    for name, group in results:
+        # TODO - how do we present errors in a slowdown plot?
+        result_dict[name] = group.groupby("WorldSize", as_index=False).mean()
+
+    return result_dict
+
+
+def _check_results(native_results, wasm_results):
+    """
+    Check that both native and faasm experiments have:
+    (1) Run with the same kernels
+    (2) Run with the same number of processes
+    """
+    # If the two experiments recorded results for different kernels, error out
+    if native_results.keys() != wasm_results.keys():
+        print(
+            "Kernels mismatch!\n - Native: {}\n - Faasm: {}".format(
+                native_results.keys(), wasm_results.keys()
+            )
+        )
+        return False
+
+    # If the experiments run with different number of processes, error out
+    for key in native_results.keys():
+        if (
+            native_results[key]["WorldSize"].count()
+            != wasm_results[key]["WorldSize"].count()
+        ):
+            print("Number of processes in kernel {} mismatch!".format(key))
+            print(
+                " - Native: {}\n - Faasm: {}".format(
+                    native_results[key], wasm_results[key]
+                )
+            )
+            return False
+
+    return True
+
+
+@task
+def plot(ctx):
+    """
+    Plot slowdown for MPI kernels
+    """
+    makedirs(PLOTS_DIR, exist_ok=True)
+
+    # Load results and sanity check
+    native_results = _read_results(NATIVE_CSV)
+    wasm_results = _read_results(WASM_CSV)
+    if not _check_results(native_results, wasm_results):
+        return
+
+    # Define the independent variables
+    # Note that we must fit all the columns for each kernel, e.g. if n is odd:
+    # ... [x - w - w /2, x - w/2] [x - w/2, x + w/2] [x + w/2, x + w + w/2] ...
+    kernels = wasm_results.keys()
+    num_procs = len(
+        wasm_results[next(iter(wasm_results))]["WorldSize"].tolist()
+    )
+    xs = np.arange(len(kernels))
+    xmin = xs[0] - 0.5
+    xmax = xs[-1] + 0.5
+    if (num_procs % 2) == 0:
+        col_width = 0.5 / (0.5 + num_procs / 2)
+        x_base = [x - (num_procs / 2 * col_width + col_width / 2) for x in xs]
+    else:
+        col_width = 0.5 / (1 + num_procs / 2)
+        x_base = [x - (num_procs / 2 * col_width) for x in xs]
+
+    # Define the dependent variables
+    fig, ax = plt.subplots()
+    y = []
+    # We group the results by kernel, but plot them world size by world size,
+    # in order to get the right x offset
+    for num_proc in range(num_procs):
+        _faasm = [
+            wasm_results[kern]["StatValue"].tolist()[num_proc]
+            for kern in wasm_results.keys()
+        ]
+        _native = [
+            native_results[kern]["StatValue"].tolist()[num_proc]
+            for kern in native_results.keys()
+        ]
+        # Each bar must be set at the midpoint of the right offset
+        x = [_x + num_proc * col_width + col_width / 2 for _x in x_base]
```

Comment:
Same comment re. variables in a list comprehension like this, no need to add leading underscores.